### ISSUE TITLE
pthread_rwlock_t implementation is missing try-lock functions (IDFGH-7685)

### ISSUE THREAD

**Issue by Spacechild1** [espressif/esp-idf#9229]
ESP-IDF has recently added support for `pthread_rwlock_t` (https://github.com/espressif/esp-idf/commit/32e34447018539522c8ffae612408ef34d395783), but strangely the try-lock functions `pthread_wrlock_tryrdlock` and `pthread_wrlock_trywrlock` are missing... I'm wondering why they have been omitted?

**Comment by 0xjakob** [comment]
@Spacechild1 

> I'm wondering why they have been omitted?

They simply haven't been implemented. The original feature request came from the Rust-community who needed only the simple functionality. I'll take a look at the code within the next few days.

**Comment by Spacechild1** [comment]
Thanks!

> to see if the implementation would be easy.

They are pretty easy to implement: instead of waiting on the condition variable, you would just unlock the resource mutex and return `EBUSY`. I can also make a PR if necessary.

**Comment by svenstaro** [comment]
Has there been any progress on this? I think this functionality would be pretty neat to have. Speaking from a Rust perspective, it's actually required to make full use of the tracing ecosystem.

**Comment by elotom** [comment]
I also encountered this problem. It would be nice to have this for the Rust ecosystem.

**Comment by 0xjakob** [comment]
The try-lock functions have been merged into master. Feel free to take a look.

**Comment by svenstaro** [comment]
Looks like https://github.com/espressif/esp-idf/commit/e7312cccea37c12bea6b9a0e6ba5c9f39762391a is still not in any release. What's the policy here? What future release would this be contained in?

**Comment by 0xjakob** [comment]
Hi @svenstaro, it has been merged to master. This means it will go into the next release (v5.2), as well as all the following releases.
